code/modules/atmospherics/components/tvalve.dm

```diff
@@ -38,130 +31,66 @@
 
 /obj/machinery/atmospherics/tvalve/on_update_icon(animation)
 	if(animation)
-		flick("tvalve[src.state][!src.state]",src)
+		flick("[base_icon_state][src.state][!src.state]",src)
 	else
-		icon_state = "tvalve[state]"
-
-/obj/machinery/atmospherics/tvalve/update_underlays()
-	if(..())
-		underlays.Cut()
-		var/turf/T = get_turf(src)
-		if(!istype(T))
-			return
-		add_underlay(T, node1, turn(dir, -180))
-
-		if(istype(src, /obj/machinery/atmospherics/tvalve/mirrored))
-			add_underlay(T, node2, turn(dir, 90))
-		else
-			add_underlay(T, node2, turn(dir, -90))
+		icon_state = "[base_icon_state][state]"
 
-		add_underlay(T, node3, dir)
+	build_device_underlays(FALSE)
 
 /obj/machinery/atmospherics/tvalve/hide(var/i)
-	update_underlays()
+	update_icon()
 
-/obj/machinery/atmospherics/tvalve/network_expand(datum/pipe_network/new_network, obj/machinery/atmospherics/pipe/reference)
-	if(reference == node1)
-		network_node1 = new_network
-		if(state)
-			network_node2 = new_network
-		else
-			network_node3 = new_network
-	else if(reference == node2)
-		network_node2 = new_network
-		if(state)
-			network_node1 = new_network
-	else if(reference == node3)
-		network_node3 = new_network
-		if(!state)
-			network_node1 = new_network
-
-	if(new_network.normal_members.Find(src))
-		return 0
+/obj/machinery/atmospherics/tvalve/proc/paired_dirs() // these two dirs are connected
+	return state ? list(turn(dir, 180), turn(dir, -90)) : list(turn(dir, 180), dir)
```

Comment:
I feel like this would be a bit easier to parse as an if/else.

Reply:
Done.